util/misc.go

```diff
@@ -53,3 +53,22 @@ func GetStack() []byte {
 	buf = buf[:stackSize]
 	return buf
 }
+
+const maxCapacity int32 = 1 << 30
+
+// NextPowerOfTwo returns next power of two size for the given target cap.
+func NextPowerOfTwo(cap int32) int32 {
```

Comment:
`1 << uint(bits.Len64(x))` will do

Reply:
it seems wrong when `x = 2, 4, 8....` o.o

Reply:
Then what about `1 << uint(bits.Len32(x-1))` ?